src/java/org/apache/cassandra/db/Directories.java

```diff
@@ -1326,6 +1386,12 @@ else if (!dir.tryCreateDirectories() && !(dir.exists() && dir.isDirectory()))
         return dir;
     }
 
+    public static Optional<File> get(File base, String... subdirs)
+    {
+        File dir = subdirs == null || subdirs.length == 0 ? base : new File(base, join(subdirs));
```

Comment:
shouldn't `null` be an assertion error there?

Reply:
i just took this from getOrCreate()